code_samples/back_office/calendar/config/custom_services.yaml

```diff
@@ -10,7 +10,7 @@ services:
             $eventType: '@App\Calendar\Holidays\EventType'
 
     App\Calendar\Holidays\EventSource:
-        class: Ibexa\Tests\Calendar\Calendar\EventSource
-        factory: [ '@App\Calendar\Holidays\EventSourceFactory','createEventSource' ]
+        class: Ibexa\Calendar\EventSource\InMemoryEventSource
```

Comment:
Hi,

Thank you for highlighting the use of a Tests class here.

We do not need to link this service class to another class than itself.

```suggestion
```

Reply:
Well, I admit I was a bit lazy when creating this PR and not tackling the "other issue". `App\Calendar\Holidays\EventSource` is no class. In this case, it is just a service name looking like a class. If you look in `App\Calendar\Holidays\EventSourceFactory::createEventSource()`, then an `Ibexa\Calendar\EventSource\InMemoryEventSource` object is returned and not a `App\Calendar\Holidays\EventSource` object.

May-be the service should be renamed to a dot-separated string.

Reply:
@ms2ad  No need to name the service as it won't be used elsewhere.
If you look at `php bin/console debug:container --tag=ibexa.calendar.event.source` none is named 😉

Reply:
@adriendupuis While I cannot run the command right now, I checked the code and every service like `Ibexa\Scheduler\Calendar\FutureHideEventSource` is an acutal class in contrast to `App\Calendar\Holidays\EventSource` which is no class (see above).

Reply:
But you're right about this not being a true class.
And https://symfony.com/doc/5.4/service_container/factories.html seems to prefer to name the service with output of the factory…
OK. Let's precise the real output class. The fake class as a service name is not problematic. But we are the lazy ones here, not you.

Reply:
That looks OK to me with the "Class name" column indicating the real class and the "Service ID" column having the same naming pattern.

```
------------------------------------------------------- ------------------------------------------------------- 
  Service ID                                              Class name                                             
 ------------------------------------------------------- ------------------------------------------------------- 
  App\Calendar\Holidays\EventSource                       Ibexa\Calendar\EventSource\InMemoryEventSource         
  Ibexa\Scheduler\Calendar\FuturePublicationEventSource   Ibexa\Scheduler\Calendar\FuturePublicationEventSource  
  Ibexa\Scheduler\Calendar\FutureHideEventSource          Ibexa\Scheduler\Calendar\FutureHideEventSource         
  Ibexa\FieldTypePage\Calendar\BlockRevealEventSource     Ibexa\FieldTypePage\Calendar\BlockRevealEventSource    
  Ibexa\FieldTypePage\Calendar\BlockHideEventSource       Ibexa\FieldTypePage\Calendar\BlockHideEventSource      
 ------------------------------------------------------- -------------------------------------------------------
 ```